airflow/providers/amazon/aws/waiters/stepfunctions.json

```diff
@@ -0,0 +1,36 @@
+{
+    "version": 2,
+    "waiters": {
+        "step_function_succeeded": {
+            "operation": "DescribeExecution",
+            "delay": 30,
+            "maxAttempts": 60,
+            "acceptors": [
+                {
+                    "matcher": "path",
+                    "argument": "status",
+                    "expected": "SUCCEEDED",
+                    "state": "success"
+                },
+                {
+                    "matcher": "error",
+                    "argument": "status",
+                    "expected": "RUNNING",
+                    "state": "retry"
+                },
```

Comment:
what is this ? It's the first time I see it in a waiter definition 🤨 
What does it do that the waiter wouldn't do without those lines ?

Reply:
This is something that I've seen used in other config files in the [botocore repo](https://github.com/boto/botocore/blob/develop/botocore/data/lambda/2015-03-31/waiters-2.json#L15). There is no specific documentation about how to write these config files, but from looking at the source code, the only states that "matter" are `success` and `failure` (by matter, I mean they are the only states that `current_state` is [compared to](https://github.com/boto/botocore/blob/develop/botocore/waiter.py#L366)). By setting the state as `retry` we are just being explicit that if we match the `expected` state, we want the waiter to retry (although it would do that anyways).

Reply:
sounds like unnecessary clutter to me...
I agree to disagree, but I generally prefer not writing code/config if it has no effect